src/fsharp/service/ServiceLexing.fsi

```diff
@@ -256,3 +258,239 @@ module Keywords =
 
     /// Keywords paired with their descriptions. Used in completion and quick info.
     val KeywordsWithDescription : (string * string) list
+
+[<Experimental("This FCS API is experimental and subject to change.")>]
+module public Lexer =
+
+    open System
+    open System.Threading
+    open FSharp.Compiler.Text
+    open FSharp.Compiler.Range
+
+    [<Flags;Experimental("This FCS API is experimental and subject to change.")>]
+    type public FSharpLexerFlags =
+        | Default                       = 0x11011
+        | LightSyntaxOn                 = 0x00001
+        | Compiling                     = 0x00010 
+        | CompilingFSharpCore           = 0x00110
+        | SkipTrivia                    = 0x01000
+        | UseLexFilter                  = 0x10000
+
+    [<RequireQualifiedAccess;Experimental("This FCS API is experimental and subject to change.")>]
+    type public FSharpSyntaxTokenKind =
+        | None
```

Comment:
I suppose you're deliberately avoiding it, but there's a TokenKind emitted by the parser generator, e.g. https://github.com/dotnet/fsharp/blob/master/src/buildtools/fsyacc/fsyaccpars.fs#L40

I think it's good to have a stable external token kind as part of the FCS API however, so the code you have looks fine

Reply:
Yes, I am deliberately avoiding it. Anything internal to the compiler shouldn't be exposed as it's subject to change more often than a public API.

Reply:
Yes, that's fine, it's about time we did this, thank you!